File: 2 CPP/Leetcode-1689-PartitioningIntoMinimumNumberOfDecibinaryNumbers.cpp
//multiple versions of the code, both are 0ms runtime on leetcode
//
class Solution {
public:
    int minPartitions(string n) {
     string s = "987654321";
     int ans = 0;
     for(char &c:s){
        if (n.find(c) != string::npos){
            int ph = c-'0';
            ans = ph;
            break;
        };
        };
        return ans;
    };

};

class Solution {
public:
    int minPartitions(string n) {
     int maxNum=0;
     for(char &c:n){
        int ph = c-'0';
        if (ph>maxNum){
            maxNum = ph;
        };
     }
     return maxNum;   
    }
};

File: 1 Python/Leetcode-1689-PartitioningIntoMinimumNumberOfDecibinaryNumbers.py
# Multiple versions of the solution sorted in an ascending order of runtime
class Solution:
    def minPartitions(self, n: str) -> int:
        for i in "987654321":
            if i in n: return int(i)

class Solution:
    def minPartitions(self, n: str) -> int:
        sieve = "987654321"
        for i in sieve:
            if i in n:
                return int(i)
                break

class Solution:
    def minPartitions(self, n: str) -> int:
        sieve = ["9","8","7","6","5","4","3","2","1"]

        for i in sieve:
            if i in n:
                return int(i)
                break

class Solution:
    def minPartitions(self, n: str) -> int:
        return int(max(n))

class Solution:
    def minPartitions(self, n: str) -> int:
        return int(sorted(n)[-1])

class Solution:
    def minPartitions(self, n: str) -> int:
        maxInt = 0
        for i in n:
            if (int(i))>maxInt:
                maxInt = int(i)
        return maxInt
